### PR TITLE
fix(runtime): support watch for components with custom tag names

### PR DESCRIPTION
**Pull request by christian-bromann**
## What is the current behavior?
If users define their component with a custom tag name, the `@Watch` decorator doesn't work.

GitHub Issue Number: #3554
STENCIL-546

## What is the new behavior?
Instead of using the component meta data we just pick the tag name for the component directly.

## Documentation

n/a

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

Added an unit test for this.

## Other information

n/a

### CODE REVIEW

File: test/wdio/setup.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ const testRequiresManualSetup =
   window.__wdioSpec__.includes('custom-elements-output-tag-class-different') ||
   window.__wdioSpec__.includes('custom-elements-delegates-focus') ||
   window.__wdioSpec__.includes('custom-elements-output') ||
-  window.__wdioSpec__.includes('global-script');
+  window.__wdioSpec__.includes('global-script') ||
+  window.__wdioSpec__.endsWith('custom-tag-name.test.tsx');
```
Comment (alicewriteswrongs) [inline test/wdio/setup.ts:17]: we need this because we don't want the normal `.define` call to happen, right?

**Review by alicewriteswrongs — CHANGES_REQUESTED** [review]
change makes sense to me! Just noticed I think one little typo in the comment and had one clarification

**Review by tanner-reits — APPROVED** [review]
LGTM

File: test/wdio/setup.ts (hunk 1 above)
Comment (christian-bromann) [reply test/wdio/setup.ts:17]: Yes, so we avoid registering all the components to the registry to test the behavior where user would do this manually.

**Review by alicewriteswrongs — APPROVED** [review]

**Pull request merged**